src/Symfony/Component/Translation/Bridge/Lokalise/LokaliseProvider.php

```diff
@@ -0,0 +1,345 @@
+        $this->endpoint = $endpoint;
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('lokalise://%s', $this->endpoint);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Lokalise API recommends sending payload in chunks of up to 500 keys per request.
+     *
+     * @see https://app.lokalise.com/api2docs/curl/#transition-create-keys-post
+     */
+    public function write(TranslatorBagInterface $translatorBag): void
+    {
+        $defaultCatalogue = $translatorBag->getCatalogue($this->defaultLocale);
+
+        if (!$defaultCatalogue) {
+            $defaultCatalogue = $translatorBag->getCatalogues()[0];
+        }
+
+        $this->ensureAllLocalesAreCreated($translatorBag);
+        $existingKeysByDomain = [];
+
+        foreach ($defaultCatalogue->getDomains() as $domain) {
+            if (!\array_key_exists($domain, $existingKeysByDomain)) {
+                $existingKeysByDomain[$domain] = [];
+            }
+
+            $existingKeysByDomain[$domain] += $this->getKeysIds(array_keys($defaultCatalogue->all($domain)), $domain);
+        }
+
+        $keysToCreate = $createdKeysByDomain = [];
+
+        foreach ($existingKeysByDomain as $domain => $existingKeys) {
+            $allKeysForDomain = array_keys($defaultCatalogue->all($domain));
+            foreach (array_keys($existingKeys) as $keyName) {
+                unset($allKeysForDomain[$keyName]);
+            }
+            $keysToCreate[$domain] = $allKeysForDomain;
+        }
+
+        foreach ($keysToCreate as $domain => $keys) {
+            $createdKeysByDomain[$domain] = $this->createKeys($keys, $domain);
+        }
+
+        $this->updateTranslations(array_merge($createdKeysByDomain, $existingKeysByDomain), $translatorBag);
+    }
+
+    public function read(array $domains, array $locales): TranslatorBag
+    {
+        $translatorBag = new TranslatorBag();
+        $translations = $this->exportFiles($locales, $domains);
+
+        foreach ($translations as $locale => $files) {
+            foreach ($files as $filename => $content) {
+                $translatorBag->addCatalogue($this->loader->load($content['content'], $locale, str_replace('.xliff', '', $filename)));
+            }
+        }
+
+        return $translatorBag;
+    }
+
+    public function delete(TranslatorBagInterface $translatorBag): void
+    {
+        $catalogue = $translatorBag->getCatalogue($this->defaultLocale);
+
+        if (!$catalogue) {
+            $catalogue = $translatorBag->getCatalogues()[0];
+        }
+
+        $keysIds = [];
+
+        foreach ($catalogue->getDomains() as $domain) {
+            $keysToDelete = [];
+            foreach (array_keys($catalogue->all($domain)) as $key) {
+                $keysToDelete[] = $key;
+            }
+            $keysIds += $this->getKeysIds($keysToDelete, $domain);
+        }
+
+        $response = $this->client->request('DELETE', sprintf('projects/%s/keys', $this->projectId), [
```

Comment:
`projects/%s` should be part of the base_uri, and the projectId property should be removed

Reply:
Thanks, it lightens the `request` calls :+1: